Optimized camera rotation in decorate mode

KS-103819380 opened the pull request:
Optimized camera rotation in decorate mode which allows the makes the camera "orbits" a center point on drag rather than rotating the camera itself, which should provide a more intuitive user experience. Only the `HouseInputManager` script is changed.

GhostEntity12 reviewed (changes requested):
There appear to be some issues with the new camera rotation system which allow the camera to get below the ground by rotating forwards. It appears to be triggered by the camera being close to the point of rotation.

KS-103819380 commented:
Update:
- Fixed player being able to rotate camera underground in Decorate mode when close to center
- Changed `Interact()` to use an interface instead
- Fixed the Door in House scene not being able to change scene
- Fixed player being able to exit out of Decorate mode while still moving a furniture

GhostEntity12 commented on Assets/Scripts/Door.cs:
```diff
@@ -0,0 +1,21 @@
+    {
+        if (sceneToLoad != null)
+        {
+            // if we are in hunting scene, call respawn in house instead
+            if (FindObjectOfType<HuntingManager>() != null)
+                HuntingManager.Instance.RespawnInHouse();
+            else
```
Are these lines actually needed? We could just drag the house scene to the field instead, preventing a need for the expensive `FindObjectOfType` call

KS-103819380 replied:
I use `FindObjectOfType` because I wanted to check whether the player is in the hunting scene or not, since switching from hunting scene to house scene will merge the hunting inventory with the permanent inventory (which is done in the `RespawnInHouse` function. The other option that I thought of was just to check the scene name, but it will break if we rename a scene without also changing the code.

GhostEntity12 replied:
Since it's a singleton, could we check instead for `HuntingManager.Instance` is `null` instead?

KS-103819380 replied:
Yeah that should work. Dont know why I havent thought abt this. Fixed.

GhostEntity12 commented on Assets/Scripts/Player/HouseInputManager.cs:
```diff
@@ -96,9 +93,9 @@ private void LateUpdate()
 		{
-			SceneManager.LoadScene("ForestTestingJames");
+			hit.transform.GetComponent<IInteractable>()?.Interact();
 		}
```
Might be better as 
```cs
if (Physics.Raycast(HouseManager.Instance.ExploreCamera.transform.position, HouseManager.Instance.ExploreCamera.transform.forward, out RaycastHit hit, playerReach) && hit.transform.TryGetComponent(out IInteractable interactable))
{
	interactable.Interact();
}
```

GhostEntity12 commented on Assets/Scripts/Player/HuntingInputManager.cs:
```diff
@@ -42,20 +42,21 @@ private void OnDisable()
-			}
-		}
-	}
+        if (Physics.Raycast(camera.transform.position, camera.transform.forward, out RaycastHit hit, 3f))
+        {
+            hit.transform.GetComponent<IInteractable>()?.Interact();
+            // if (hit.transform.TryGetComponent(out Shootable shootable) && shootable.IsDead)
+            // {
+            // 	HuntingManager.Instance.HuntingInventory.AddItem(shootable.GetInventoryItem());
+            // 	Debug.Log(HuntingManager.Instance.HuntingInventory);
+            // 	Destroy(shootable.gameObject);
+            // }
+            // //if we are interacting with a door, load the house scene
+            // else if (hit.transform.gameObject.layer == LayerMask.NameToLayer("Door"))
+            // {
+            // 	HuntingManager.Instance.RespawnInHouse();
+            // }
+        }
```
Same thing here
```cs 
if (Physics.Raycast(camera.transform.position, camera.transform.forward, out RaycastHit hit, 3f) && hit.transform.TryGetComponent(out IInteractable interactable))
{
	interactable.Interact();
}
```

GhostEntity12 reviewed (commented):
Looks good, want to get your opinion on the comments, but will approve after
